lib/nvme/nvme_pcie.c

```diff
@@ -1841,7 +1841,14 @@ nvme_pcie_qpair_check_timeout(struct spdk_nvme_qpair *qpair)
 		 * Request has timed out. This could be i/o or admin request.
 		 * Call the registered timeout function for user to take action.
 		 */
-		ctrlr->timeout_cb_fn(ctrlr, qpair, ctrlr->timeout_cb_arg);
+
+		tr->req->timeout_count++;
+		tr->submit_tick = spdk_get_ticks() + ctrlr->timeout_ticks;
```

Comment:
This should probably be resetting submit_tick to the current spdk_get_ticks() - the timeout_ticks addition should be moved to the place that is checking for timeout. (We modified the way timeouts are checked so that submission time can be recorded separately and used for I/O statistics.)